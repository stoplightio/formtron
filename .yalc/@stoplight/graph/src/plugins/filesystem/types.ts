import { Stats } from 'fs';

import { INodeInstance, INodeOptions } from '../../types';

export enum FilesystemTypes {
  DIRECTORY = 'directory',
  FILE = 'file',
}

export interface IFilesystemPluginOpts {
  recursive?: boolean;
}

export interface IDirectoryInput extends INodeOptions<IDirectoryContent> {
  type: FilesystemTypes.DIRECTORY;
  path: string;
  stats?: Stats;
}

export interface IDirectory extends INodeInstance<IDirectoryContent> {
  type: FilesystemTypes.DIRECTORY;
  path: string;
  name: string;
  dirname: string;
}

export type IDirectoryContent = Array<{
  name: string;
  path: string;
  stats: Stats;
}>;

export type IFileContent = string;

export interface IFileInput extends INodeOptions<IFileContent> {
  type: FilesystemTypes.FILE;
  path: string;
  stats?: Stats;
}

export interface IFile extends INodeInstance<IFileContent> {
  type: FilesystemTypes.FILE;
  path: string;
  name: string;
  fileName: string;
  dirname: string;
  contentType: string;
  stats?: Stats;
}
