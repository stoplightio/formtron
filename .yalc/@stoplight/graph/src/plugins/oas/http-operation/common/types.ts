import {
  IApiKeySecurityScheme,
  IBasicSecurityScheme,
  IBearerSecurityScheme,
  IOauth2SecurityScheme,
  IOpenIdConnectSecurityScheme,
} from '@stoplight/types';
import { OpenAPIObject, OperationObject, PathObject } from 'openapi3-ts';
import { Operation, Path, Spec } from 'swagger-schema-official';

export declare type HttpSecurityScheme =
  | IApiKeySecurityScheme
  | IBearerSecurityScheme
  | IBasicSecurityScheme
  | IOauth2SecurityScheme
  | IOpenIdConnectSecurityScheme;

export interface ITranslateOperationOptions<O, P, S> {
  operation: O;
  method: string;
  pathObject: P;
  path: string;
  spec: S;
}

export type Oas2TranslateOperationOptions = ITranslateOperationOptions<Operation, Path, Spec>;
export type Oas3TranslateOperationOptions = ITranslateOperationOptions<OperationObject, PathObject, OpenAPIObject>;
