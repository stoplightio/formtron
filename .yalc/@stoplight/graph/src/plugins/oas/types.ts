export enum Oas2RootNodeTypes {
  OAS2 = 'oas2',
  SECURITY_SCHEMES = 'oas2_security_schemes',
  PATHS = 'oas2_paths',
  TAGS = 'oas2_tags',
  SHARED = 'oas2_shared',
}

export enum Oas2ChildNodeTypes {
  SECURITY_SCHEME = 'oas2_security_scheme',
  PATH = 'oas2_path',
  TAG = 'oas2_tag',
}

type OasBaseParamType = 'header' | 'query' | 'path';

export type Oas2ParamType = OasBaseParamType | 'body';
export type Oas3ParamType = OasBaseParamType | 'cookie';

export enum Oas3RootNodeTypes {
  OAS3 = 'oas3',
  SHARED = 'oas3_shared',
  TAGS = 'oas3_tags',
  SERVERS = 'oas3_servers',
  INFO = 'oas3_info',
  OPERATION = 'operation',
  COMPONENTS = 'oas3_components',
  PATHS = 'oas3_paths',
}

export enum Oas3ChildNodeTypes {
  SECURITY_SCHEME = 'oas3_security_scheme',
  PATH = 'oas3_path',
  TAG = 'oas3_tag',
  SERVER = 'oas3_server',
}

export enum OasNodeChildType {
  OBJECT = 1,
  ARRAY,
}

export enum Oas2SharedNodeTypes {
  PARAMETERS = 'oas2_parameters',
  PARAMETER = 'oas2_parameter',
  RESPONSES = 'oas2_responses',
  RESPONSE = 'oas2_response',
  JSON_SCHEMA = 'json_schema',
  SCHEMAS = 'oas2_schemas',
  OPERATION = 'oas2_operation',
}

export enum Oas3SharedNodeTypes {
  SCHEMAS = 'oas3_schemas',
  RESPONSES = 'oas3_responses',
  RESPONSE = 'oas3_response',
  PARAMETERS = 'oas3_parameters',
  EXAMPLES = 'oas3_examples',
  REQUEST_BODIES = 'oas3_request_bodies',
  HEADERS = 'oas3_headers',
  SECURITY_SCHEMES = 'oas3_security_schemes',
  LINKS = 'oas3_links',
  CALLBACKS = 'oas3_callbacks',
  OPERATION = 'oas3_operation',
}
