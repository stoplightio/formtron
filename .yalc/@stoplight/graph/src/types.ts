import { IParserResult } from '@stoplight/types';

export type IGraphLoader<Node extends INodeInstance = INodeInstance, NodeContent = any> = (
  node: Node
) => Promise<NodeContent>;

export interface IGraphHookApi<Input, Output, Parent> {
  createNode: (node: Input, opts?: { parent?: Parent }) => Promise<Output>;
  createEdge: (source: INodeInstance, destination: INodeInstance) => void;
  getNode: (id: string) => INodeInstance | undefined;
  getParentNode: (node: INodeInstance) => INodeInstance | undefined;
}

export interface IGraphHook<
  Input extends INodeOptions = INodeOptions,
  Output extends INodeInstance = INodeInstance,
  Parent extends INodeInstance = INodeInstance
> {
  name: string;

  // by default node type is set to name, but can override here if necessary
  // TODO(SL-248): do we even need this? It looks that this is only used to ensure that the produced node
  // is of the same type that what hook declares. Can't we solve that differently?
  nodeType?: string;

  // allow plugins to opt into dangerous recursive handling
  // we can put additional safeguards around this later
  recursive?: boolean;

  // scopes which nodes this plugin is interested in
  selector?: ((node: INodeInstance) => boolean) | string;

  // events
  onWillCreateNode?: (node: Partial<Output>) => void;

  // TODO(SL-248): why does that have to return anything? What is `any`?
  onDidCreateNode?: (parent: Parent, opts: IGraphHookApi<Input, Output, Parent>) => Promise<any>;
}

export interface IGraphPlugin {
  hooks?: Array<IGraphHook<any, any, any>>;
  loaders?: {
    [nodeType: string]: IGraphLoader<any, any>;
  };
}

/**
 * Nodes
 */

export enum NodeTypes {
  PARSED = 'parsed',
}

// When creating a node.
export interface INodeOptions<Content = unknown> extends Object {
  id: string;
  type: string;
  contentType?: string;
  // TODO(SL-248): I think this should be private in context of INode because INode has loadContent function
  content?: Content;
}

// When dealing with a node instance.
export interface INodeInstance<Content = unknown> extends INodeOptions<Content> {
  loadContent: () => Promise<Content>;
}

export interface IParsedNode<T> extends INodeInstance<IParserResult<T>> {
  type: NodeTypes.PARSED;
}
