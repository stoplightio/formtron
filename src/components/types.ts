export enum Variant {
  normal = '',
  invalid = 'invalid',
}
