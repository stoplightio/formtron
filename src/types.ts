import * as React from 'react';
import { Dictionary } from 'ts-essentials';
import { themeTypes } from './theme';

export interface IFormtronCommon {
  id?: string;
  value: any;
  onChange: (value: any) => void;

  schema: any;
  fieldComponents: Dictionary<React.FunctionComponent<IFormtronControl>>;
}

export interface IFormtronControl extends IFormtronCommon {
  path: string[];
}

export interface IFormtron extends IFormtronCommon {
  selection: string;
  onInternalChange?: Function;
  themeName?: themeTypes;
}

export interface IAddOperation {
  op: 'add';
  path: string;
  value: any;
}

export interface IMoveOperation {
  op: 'move';
  from: string;
  path: string;
}

export interface ISelectOperation {
  op: 'select';
  from: string;
  path: string;
}

export type IOperation = IAddOperation | IMoveOperation | ISelectOperation;
