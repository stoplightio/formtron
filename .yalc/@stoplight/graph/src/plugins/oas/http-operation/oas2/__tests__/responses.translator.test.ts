import { IHttpHeaderParam } from '@stoplight/types';
import { translateToHeaderParams } from '../helpers/params.translator';
import { translateToResponses } from '../helpers/responses.translator';
jest.mock('../helpers/params.translator');

describe('responses.translator', () => {
  const fakeHeaderParams: IHttpHeaderParam[] = [{ name: 'fake-header', content: {} }];
  const produces = ['*'];

  beforeEach(() => {
    (translateToHeaderParams as jest.Mock).mockReturnValue(fakeHeaderParams);
  });
  test('should translate to multiple responses', () => {
    const responses = translateToResponses(
      {
        r1: {
          description: 'd1',
          examples: {
            e1: {},
          },
          headers: {},
          schema: {},
        },
        r2: {
          description: 'd2',
          examples: {
            e2: {},
          },
          headers: {},
          schema: {},
        },
      },
      produces
    );

    expect(responses).toMatchSnapshot();
  });

  test('should translate to response w/o headers', () => {
    expect(
      translateToResponses(
        {
          r1: {
            description: 'd1',
            examples: {
              e1: {},
            },
            schema: {},
          },
        },
        produces
      )
    ).toMatchSnapshot();
  });

  test('should translate to response w/o examples', () => {
    expect(
      translateToResponses(
        {
          r1: {
            description: 'd1',
            schema: {},
          },
        },
        produces
      )
    ).toMatchSnapshot();
  });
});
