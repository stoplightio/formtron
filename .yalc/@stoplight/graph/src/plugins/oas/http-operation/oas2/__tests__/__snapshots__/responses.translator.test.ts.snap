// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responses.translator should translate to multiple responses 1`] = `
Array [
  Object {
    "code": "r1",
    "contents": Array [
      Object {
        "examples": Array [
          Object {
            "key": "e1",
            "value": Object {},
          },
        ],
        "mediaType": "*",
        "schema": Object {},
      },
    ],
    "description": "d1",
    "headers": Array [
      Object {
        "content": Object {},
        "name": "fake-header",
      },
    ],
  },
  Object {
    "code": "r2",
    "contents": Array [
      Object {
        "examples": Array [
          Object {
            "key": "e2",
            "value": Object {},
          },
        ],
        "mediaType": "*",
        "schema": Object {},
      },
    ],
    "description": "d2",
    "headers": Array [
      Object {
        "content": Object {},
        "name": "fake-header",
      },
    ],
  },
]
`;

exports[`responses.translator should translate to response w/o examples 1`] = `
Array [
  Object {
    "code": "r1",
    "contents": Array [
      Object {
        "examples": Array [],
        "mediaType": "*",
        "schema": Object {},
      },
    ],
    "description": "d1",
    "headers": Array [
      Object {
        "content": Object {},
        "name": "fake-header",
      },
    ],
  },
]
`;

exports[`responses.translator should translate to response w/o headers 1`] = `
Array [
  Object {
    "code": "r1",
    "contents": Array [
      Object {
        "examples": Array [
          Object {
            "key": "e1",
            "value": Object {},
          },
        ],
        "mediaType": "*",
        "schema": Object {},
      },
    ],
    "description": "d1",
    "headers": Array [
      Object {
        "content": Object {},
        "name": "fake-header",
      },
    ],
  },
]
`;
